lib/dialogue-agent/entity-linking/entity-finder.js

```diff
@@ -70,7 +70,7 @@ function getBestEntityMatch(searchTerm, entityType, candidates) {
         for (let candToken of candTokens) {
             let found = false;
             for (let token of searchTermTokens) {
-                if (editDistance(token, candToken) <= 1) {
+                if (editDistance(token, candToken) <= 1 && token.length > 1 || editDistance(token, candToken) === 0) {
```

Comment:
editDistance() == 0 can be written a lot faster as `token === candToken`

Reply:
Also, for clarity please put parenthesis around the first clause of ||.
Also, for speed you should always write your conditions from faster to slower, because logic operators are short-circuited.

Reply:
Wait did you revert the fix for this in some later comment? Please check that you didn't make mistakes in rebasing this PR. Maybe do a full rebase and squash, force push and look at it before merging.